src/emd/rt_bse_types.F

```diff
@@ -247,13 +247,23 @@ MODULE rt_bse_types
       ! Workspace required for hartree_pw
       TYPE(dbcsr_type)                                          :: v_dbcsr = dbcsr_type(), &
                                                                    w_dbcsr = dbcsr_type()
+#if defined(__GNUC__) && defined(__GNUC_MINOR__) && CPVERSION(9, 5) > CPVERSION(__GNUC__, __GNUC_MINOR__)
+      TYPE(dbt_type)                                            :: screened_dbt, &
+                                                                   sigma_dbt, &
+                                                                   greens_dbt, &
+                                                                   t_3c_w, &
+                                                                   t_3c_work_RI__AO_AO, &
+                                                                   t_3c_work_RI_AO__AO, &
+                                                                   t_3c_work2_RI_AO__AO
+#else
       TYPE(dbt_type)                                            :: screened_dbt = dbt_type(), &
                                                                    sigma_dbt = dbt_type(), &
                                                                    greens_dbt = dbt_type(), &
                                                                    t_3c_w = dbt_type(), &
                                                                    t_3c_work_RI__AO_AO = dbt_type(), &
                                                                    t_3c_work_RI_AO__AO = dbt_type(), &
                                                                    t_3c_work2_RI_AO__AO = dbt_type()
+#endif
```

Comment:
Many thanks to everyone for the discussion - I included the following preprocessor if statement in emd/rt_bse_types.F and a similar one in post_scf_bandstructure_types.F.
```suggestion
#if defined(__GNUC__) && defined(__GNUC_MINOR__) && CPVERSION(9, 5) > CPVERSION(__GNUC__, __GNUC_MINOR__)
      TYPE(dbt_type)                                            :: screened_dbt, &
                                                                   sigma_dbt, &
                                                                   greens_dbt, &
                                                                   t_3c_w, &
                                                                   t_3c_work_RI__AO_AO, &
                                                                   t_3c_work_RI_AO__AO, &
                                                                   t_3c_work2_RI_AO__AO
#else
      TYPE(dbt_type)                                            :: screened_dbt = dbt_type(), &
                                                                   sigma_dbt = dbt_type(), &
                                                                   greens_dbt = dbt_type(), &
                                                                   t_3c_w = dbt_type(), &
                                                                   t_3c_work_RI__AO_AO = dbt_type(), &
                                                                   t_3c_work_RI_AO__AO = dbt_type(), &
                                                                   t_3c_work2_RI_AO__AO = dbt_type()
#endif
```
Let me know whether this is ok, I will also have to wait for the checks to rerun.